docs/spec_draft.md

```diff
@@ -1444,6 +1445,144 @@ specification. Numeric precision is implementation-defined.
+    where:
+     * `eval(node)` = `scatter(eval(node.left), eval(node.right))`.
+     * `eval(leaf)` = `leaf.value`.
+
+   * `schedule` is an implementation-defined full binary tree whose in-order
+     traversal consists of:
+     * `source_values[index]` values, for all `index` in the index space
+      of `source_values`, in the ascending lexicographic order of `index`.
+      * Interspersed with an implementation-defined amount of `init_value`
+      at implementation-defined positions.
+
+### Inputs
+
+| Name                | Type                                       | Constraints      |
+|---------------------|--------------------------------------------|------------------|
+| `operand`           | tensor of any supported type               | (C1-C9)          |
+| `source`            | tensor of any supported type               | (C1-C3)          |
+| `init_value`        | 0-dimensional tensor of any supported type |                  |
+| `window_dimensions` | 1-dimensional tensor constant type `si64`  | (C1), (C3), (C4) |
+| `window_strides`    | 1-dimensional tensor constant type `si64`  | (C3), (C5)       |
+| `padding`           | 2-dimensional tensor constant type `si64`  | (C3), (C6)       |
+| `select`            | `function`                                 | (C7)             |
+| `scatter`           | `function`                                 | (C4), (C8)       |
+
+### Outputs
+
+| Name     | Type                         | Constraints |
+|----------|------------------------------|-------------|
+| `result` | tensor of any supported type | (C9)        |
+
+### Constraints
+
+  * (C1) rank(`operand`) $=$ rank(`source`) $=$ size(`window_dimensions`).
+
+  * (C2) `operand` and `source` have the same element type.
+
+  * (C3) `dim(source, i) = strided_bound[i]` for any i $\in$ axis(`source`)
+    * `strided_bound[i] = (padded_operand[i] == 0 || window_dimensions[i] > padded_operand[i]) ? 0 : floor((padded_operand[i] - window_dimensions[i]) / window_strides[i]) + 1`.
+    * `padded_operand[i] = padding[i][0] + dim(operand, i) + padding[i][1]`.
+
+  * (C4) On `window_dimensions`
+    * size(`window_dimensions`) $=$ rank(`operand`).
+    * `window_dimensions[i]` $\gt 0$ for all i $\in$ [0, size(window_dimensions)).
+
+  * (C5) On `window_strides` (optional)
+    * Default value of 1 for each dimension of `operand`.
+    * size(`window_strides`) $=$ rank(`operand`).
+    * `window_strides[i]` $\gt 0$ for all i $\in$ [0, size(window_strides)).
+
+  * (C6) On `padding`(optional)
+    * Default value of (0,0) for each dimension of `operand`.
+    * dim(`padding`, 0) $=$ rank(`operand`) and dim(`padding`, 1) = 2.
+
+  * (C7) `select` has type `(tensor<E>, tensor<E>) -> tensor<E>` where
+         `E = element_type(operand)`.
+
+  * (C8) The `scatter` has type `(tensor<E>, tensor<E>) -> tensor<E>` where
+         `E = element_type(operand)`.
+
+  * (C9) type(`operand`) $=$ type(`result`).
```

Comment:
Also, I don't remember what PR it was, but I think we decided to not make default values part of the spec and treat them as part of the API. Given that, let's drop "(optional)" and "Default value...".